zebrad/src/components/tokio.rs

```diff
@@ -22,4 +23,18 @@ impl TokioComponent {
             rt: Some(Runtime::new().unwrap()),
         })
     }
+
+    pub fn start_signal_handler(&self) {
+        self.rt
+            .as_ref()
+            .expect("this option is only to work around locks, runtime should be here")
+            .spawn(signal_handler());
+    }
+}
+
+/// Zebrad's handler for various signals
+async fn signal_handler() {
```

Comment:
Does this handle signals besides a SIGINT ?

Reply:
Not yet, I'm going to copy the signal handling code from `linkerd2_proxy` for this, just wanted to get the reorg stuff done and pushed first.

Reply:
Actually looking at the linkerd code the only signal they additionally handle is SIGTERM and they do it for a specific reason. https://github.com/linkerd/linkerd2-proxy/blob/1b3e899efad2ad4ee471c5b49482437e7a325bde/linkerd/signal/src/lib.rs

I'm leaning towards only handling ctrl-c for now and leaving everything else as it already was given that `ctrl-c` was the only thing preventing me from getting the flamegraph when I was testing this. I think it is sufficient for the situations we'll be using the flamegraph generation for and if it isn't we can fix things as the need arises.